connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala

```diff
@@ -92,9 +92,13 @@ object SchemaConverters {
             MapType(keyType, valueType, valueContainsNull = false).defaultConcreteType,
             nullable = false))
       case MESSAGE =>
+        // Stop recursion at the first level when a recursive field is encountered.
+        // TODO: The user should be given the option to set the recursion level to 1, 2, or 3
```

Comment:
@rangadi Instead of limiting the recursion to only one level, the user should be able to choose a recursion level of 1, 2, or 3. Going beyond 3 levels of recursion should not be allowed. any thoughts? 

`spark.protobuf.recursion.level`

Reply:
Yeah, I think it is useful. Users may not be able to remove recursive references, but might be willing to limit recursion.
I think the default should be an error with a clear message about how users can set configuration. 
Also, I don't think it should be spark config, but rather an `option` passed in.

Reply:
Are you planning to add selectable recursion depth here or in a follow up?

Reply:
@rangadi planning to add the selectable recursion depth in this PR.